src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -0,0 +1,333 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import io.confluent.connect.avro.AvroData;
+import io.confluent.connect.hdfs.filter.CommittedFileFilter;
+import io.confluent.connect.hdfs.filter.TopicCommittedFileFilter;
+import io.confluent.connect.hdfs.hive.HiveMetaStore;
+import io.confluent.connect.hdfs.hive.HiveUtil;
+import io.confluent.connect.hdfs.partitioner.Partitioner;
+import io.confluent.connect.hdfs.storage.Storage;
+import io.confluent.connect.hdfs.storage.StorageFactory;
+
+public class DataWriter {
+  private static final Logger log = LoggerFactory.getLogger(DataWriter.class);
+
+  private Map<TopicPartition, TopicPartitionWriter> topicPartitionWriters;
+  private String url;
+  private Storage storage;
+  private Configuration conf;
+  private String topicsDir;
+  private Format format;
+  private Set<TopicPartition> assignment;
+  private Partitioner partitioner;
+  private RecordWriterProvider writerProvider;
+  private SchemaFileReader schemaFileReader;
+  private Map<TopicPartition, Long> offsets;
+  private Set<TopicPartition> lastAssignment;
+  private HdfsSinkConnectorConfig connectorConfig;
+  private AvroData avroData;
+  private SinkTaskContext context;
+  private ExecutorService executorService;
+  private String hiveDatabase;
+  private HiveMetaStore hiveMetaStore;
+  private HiveUtil hive;
+  private Queue<Future> futures;
+  private boolean hiveIntegration;
+
+  @SuppressWarnings("unchecked")
+  public DataWriter(HdfsSinkConnectorConfig connectorConfig, SinkTaskContext context, AvroData avroData) {
+    try {
+      this.connectorConfig = connectorConfig;
+      this.avroData = avroData;
+      this.context = context;
+
+      conf = new Configuration();
+      url = connectorConfig.getString(HdfsSinkConnectorConfig.HDFS_URL_CONFIG);
+      topicsDir = connectorConfig.getString(HdfsSinkConnectorConfig.TOPICS_DIR_CONFIG);
+      String logsDir = connectorConfig.getString(HdfsSinkConnectorConfig.LOGS_DIR_CONFIG);
+
+      Class<? extends Storage> storageClass = (Class<? extends Storage>) Class
+              .forName(connectorConfig.getString(HdfsSinkConnectorConfig.STORAGE_CLASS_CONFIG));
+      storage = StorageFactory.createStorage(storageClass, conf, url);
+
+      createDir(topicsDir);
+      createDir(logsDir);
+
+      format = getFormat();
+      writerProvider = format.getRecordWriterProvider();
+      schemaFileReader = format.getSchemaFileReader(avroData);
+
+      partitioner = createPartitioner(connectorConfig);
+
+      assignment = context.assignment();
+      offsets = new HashMap<>();
+      lastAssignment = new HashSet<>();
+
+      hiveIntegration = connectorConfig.getBoolean(HdfsSinkConnectorConfig.HIVE_INTEGRATION_CONFIG);
```

Comment:
If the DB is not provided, the tables will be created in the Hive default database. We need both configurations.